plugins/woocommerce/src/Blocks/Domain/Services/Notices.php

```diff
@@ -68,6 +72,28 @@ public function add_kses_notice_allowed_tags( $allowed_tags ) {
 		return array_merge( $allowed_tags, $svg_args );
 	}
 
+	/**
+	 * Replaces core notice templates with those from blocks.
+	 *
+	 * The new notice templates match block components with matching icons and styling. The only difference is that core
+	 * only has notices for info, success, and error notices, whereas blocks has notices for info, success, error,
+	 * warning, and a default notice type.
```

Comment:
I'd comment here that we are using different CSS classes also as a difference. 

Also, for block themes we introduced these 2 new notice types: warning and default; but they don't have any templates yet. I did not see requests for them, but as they are a default on the block themes we might need to add them in the future. What do you think @opr ?

Reply:
> Also, for block themes we introduced these 2 new notice types: warning and default; but they don't have any templates yet. I did not see requests for them, but as they are a default on the block themes we might need to add them in the future. What do you think @opr ?

I'm not 100% sure so take this with a grain of salt but I _think_ because of the contexts where these new notice types appear the PHP templates would not customise them. The templates are only customising "classic" notice types (info, error, success) but the new types only show in new blocks we've added, where the block style notices appear. E.g. in the Checkout **block**, the templates would not override these notices as they are not taken from the PHP files.

Does that make sense, or have I misunderstood the question?

Reply:
The new types of notices are to be activated in block themes, also outside of the checkout blocks:
Look at the My account section on your branch

Block theme 
<img width="1059" alt="Screenshot 2024-02-02 at 13 40 52" src="https://github.com/woocommerce/woocommerce/assets/1628454/6e3634cc-e726-4da9-b294-368ccfe547fb">


Storefront
<img width="804" alt="Screenshot 2024-02-02 at 13 41 26" src="https://github.com/woocommerce/woocommerce/assets/1628454/a885c6ae-de3f-4678-9cb6-170a22b62847">

Reply:
Yup, but this is an `info` notice which already existed, so is drawn from the info template. What I mean is no `warning` or `default` notices show up in these locations

Reply:
I've created an issue https://github.com/woocommerce/woocommerce/issues/44324